Test/Core/Components/RigidBodyTest.cs

```diff
@@ -16,6 +13,46 @@ namespace Duality.Tests.Components
 	[TestFixture]
 	public class RigidBodyTest
 	{
+		[Test] public void FallingBallOnPlatform()
+		{
+			Scene scene = new Scene();
+			Scene.SwitchTo(scene);
+
+			// Create the ball
+			GameObject ball = new GameObject("Ball");
+			Transform ballTransform = ball.AddComponent<Transform>();
+			RigidBody ballBody = ball.AddComponent<RigidBody>();
+			CollisionEventReceiver listener = ball.AddComponent<CollisionEventReceiver>();
+			ballBody.AddShape(new CircleShapeInfo(1, new Vector2(0, 0), 1));
+			scene.AddObject(ball);
+
+			// Create the platform
+			GameObject platform = new GameObject("Platform");
+			Transform platformTrans = platform.AddComponent<Transform>();
+			platformTrans.Pos = new Vector3(0, 5, 0);
+			RigidBody platformBody = platform.AddComponent<RigidBody>();
+			platformBody.AddShape(new ChainShapeInfo(new[] { new Vector2(-1, 0), new Vector2(1, 0), }));
+			platformBody.BodyType = BodyType.Static;
+			scene.AddObject(platform);
+
+			// Simulate some update steps
+			for (int i = 0; i < 6; i++)
+			{
+				DualityApp.Update(true);
+			}
+
+			// Check if the position is within expected values
+			Assert.IsTrue(ballTransform.Pos.Y > 3f);
+			Assert.IsTrue(ballTransform.Pos.Y < 5f);
+			Assert.IsTrue(Math.Abs(ballTransform.Pos.X) < 0.01f);
+			Assert.IsTrue(Math.Abs(ballTransform.Pos.Z) < 0.01f);
+
+			// Check if the collision events were triggered
+			Assert.IsTrue(listener.CollisionBegin.Count == 1);
+			Assert.IsTrue(listener.CollisionEnd.Count == 0); //The ball should keep touching the platform once it collides
```

Comment:
👍 for the explaining comment, but there's that missing space again 😄

Reply:
Note to self: write space inserting comment AI